inference-engine/src/transformations/src/transformations/common_optimizations/eliminate_unsqueeze_gather.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/eliminate_unsqueeze_gather.hpp"
+
+#include <ngraph/opsets/opset6.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+#include "itt.hpp"
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::EliminateUnsqueezeGather, "EliminateUnsqueezeGather", 0);
+
+ngraph::pass::EliminateUnsqueezeGather::EliminateUnsqueezeGather() {
+    MATCHER_SCOPE(EliminateUnsqueezeGather);
+    const auto unsqueezeAxis = ngraph::pattern::any_input();
+    const auto unsqueeze = ngraph::pattern::wrap_type<ngraph::opset6::Unsqueeze>({ngraph::pattern::any_input(), unsqueezeAxis});
+    const auto gatherIndices = ngraph::opset6::Constant::create(ngraph::element::i64, ngraph::Shape{}, {0});
+    const auto gatherAxis = ngraph::pattern::any_input();
+    const auto gather = ngraph::pattern::wrap_type<ngraph::opset6::Gather>({unsqueeze, gatherIndices, gatherAxis});
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+        auto& patternValue = m.get_pattern_value_map();
+
+        const auto& m_unsqueezeAxis = patternValue.at(unsqueezeAxis);
+        const auto& m_gatherAxis = patternValue.at(gatherAxis);
+
+        const auto& unsqueezeAxisNode = ngraph::as_type_ptr<ngraph::opset6::Constant>(m_unsqueezeAxis.get_node_shared_ptr());
+        const auto& gatherAxisNode = ngraph::as_type_ptr<ngraph::opset6::Constant>(m_gatherAxis.get_node_shared_ptr());
+
+        if (!unsqueezeAxisNode || !gatherAxisNode) {
+            return false;
+        }
+
+        const auto& unsqueezeAxisVec = unsqueezeAxisNode->cast_vector<int64_t>();
+        const auto& gatherAxisVec = gatherAxisNode->cast_vector<int64_t>();
+
+        if (unsqueezeAxisVec.size() != 1 || gatherAxisVec.size() != 1) {
```

Comment:
Please add comments that explains the situation where we can remove this unsqueeze-gather